src/Workspaces/Core/MSBuild/GlobalSuppressions.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+// This file is used by Code Analysis to maintain SuppressMessage
+// attributes that are applied to this project.
+// Project-level suppressions either have no target or are given
+// a specific target and scoped to a namespace, type, member, etc.
+
+using System.Diagnostics.CodeAnalysis;
+
+[assembly: SuppressMessage("ApiDesign", "RS0026:Do not add multiple public overloads with optional parameters", Justification = "Special case to avoid ILogger type getting loaded in downstream clients", Scope = "member", Target = "~M:Microsoft.CodeAnalysis.MSBuild.MSBuildWorkspace.OpenSolutionAsync(System.String,System.IProgress{Microsoft.CodeAnalysis.MSBuild.ProjectLoadProgress},System.Threading.CancellationToken)~System.Threading.Tasks.Task{Microsoft.CodeAnalysis.Solution}")]
```

Comment:
💡 Move all these suppressions to the method(s) they apply to

Reply:
Is that the official policy for the IDE team? I typically only put suppression messages in the code file if we indent to remove them one day. As is stands this situation is expected to be permanent, so what is the advantage of polluting the code file with these suppressions? They are not something future maintainers are expected to reason about correct?

Reply:
Yeah we don't use GlobalSuppressions.cs. Just #pragma disable and comment that as needed.